src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -486,6 +496,10 @@ public function merge(ContainerBuilder $container)
         $this->addAliases($container->getAliases());
         $this->getParameterBag()->add($container->getParameterBag()->all());
 
+        if (!$this->context->isLocked()) {
+            $this->context->merge($container->getContext());
```

Comment:
Perhaps always merge (remove the if), and depend on `isFrozen()`? Which currently locks the context, ie. this can never happen (having a non-locked context) :)